src/plugins/explore/public/application/utils/state_management/actions/query_actions.ts

```diff
@@ -285,67 +285,76 @@ export const executeQueries = createAsyncThunk<
     );
   }
 
+  // Wait for main queries (data table and histogram) to complete
+  await Promise.all(promises);
```

Comment:
are we sure we want to do this? One of the things we wanted before was we didn't want histogram to block the rest of UI. They should be able to be loaded asynchrounously